rules/terraformrules/terraform_module_semver_source_test.go

```diff
@@ -0,0 +1,383 @@
+			Content: `
+module "pinned_git" {
+  source = "git://hashicorp.com/consul.git?ref=1.2.3"
+}`,
+			Expected: tflint.Issues{},
+		},
+		{
+			Name: "github module is not pinned",
+			Content: `
+module "unpinned" {
+  source = "github.com/hashicorp/consul"
+}`,
+			Expected: tflint.Issues{
+				{
+					Rule:    NewTerraformModuleSemverSourceRule(),
+					Message: "Module source \"github.com/hashicorp/consul\" is not pinned",
+					Range: hcl.Range{
+						Filename: "module.tf",
+						Start:    hcl.Pos{Line: 3, Column: 12},
+						End:      hcl.Pos{Line: 3, Column: 41},
+					},
+				},
+			},
+		},
+		{
+			Name: "github module reference is not semver",
+			Content: `
+module "default_git" {
+  source = "github.com/hashicorp/consul.git?ref=master"
+}`,
+			Expected: tflint.Issues{
+				{
+					Rule:    NewTerraformModuleSemverSourceRule(),
+					Message: "Module source \"github.com/hashicorp/consul.git?ref=master\" uses a ref which is not a version string",
+					Range: hcl.Range{
+						Filename: "module.tf",
+						Start:    hcl.Pos{Line: 3, Column: 12},
+						End:      hcl.Pos{Line: 3, Column: 56},
+					},
+				},
+			},
+		},
+		{
+			Name: "github module reference is semver",
+			Content: `
+module "pinned_git" {
+  source = "github.com/hashicorp/consul.git?ref=v1.2.3"
+}`,
+			Expected: tflint.Issues{},
+		},
+		{
+			Name: "github module reference is semver (no leading v)",
+			Content: `
+module "pinned_git" {
+  source = "github.com/hashicorp/consul.git?ref=v1.2.3"
+}`,
+			Expected: tflint.Issues{},
+		},
+		{
+			Name: "bitbucket module is not pinned",
+			Content: `
+module "unpinned" {
+  source = "bitbucket.org/hashicorp/consul"
+}`,
+			Expected: tflint.Issues{
+				{
+					Rule:    NewTerraformModuleSemverSourceRule(),
+					Message: "Module source \"bitbucket.org/hashicorp/consul\" is not pinned",
+					Range: hcl.Range{
+						Filename: "module.tf",
+						Start:    hcl.Pos{Line: 3, Column: 12},
+						End:      hcl.Pos{Line: 3, Column: 44},
+					},
+				},
+			},
+		},
+		{
+			Name: "bitbucket git module reference is not semver",
+			Content: `
+module "default_git" {
+  source = "bitbucket.org/hashicorp/consul.git?ref=master"
+}`,
+			Expected: tflint.Issues{
+				{
+					Rule:    NewTerraformModuleSemverSourceRule(),
+					Message: "Module source \"bitbucket.org/hashicorp/consul.git?ref=master\" uses a ref which is not a version string",
+					Range: hcl.Range{
+						Filename: "module.tf",
+						Start:    hcl.Pos{Line: 3, Column: 12},
+						End:      hcl.Pos{Line: 3, Column: 59},
+					},
+				},
+			},
+		},
+		{
+			Name: "bitbucket git module reference is pinned to semver",
+			Content: `
+module "pinned_git" {
+  source = "bitbucket.org/hashicorp/consul.git?ref=v1.2.3"
+}`,
+			Expected: tflint.Issues{},
+		},
+		{
+			Name: "bitbucket git module reference is pinned to semver (no leading v)",
+			Content: `
+module "pinned_git" {
+  source = "bitbucket.org/hashicorp/consul.git?ref=v1.2.3"
```

Comment:
```suggestion
  source = "bitbucket.org/hashicorp/consul.git?ref=1.2.3"
```

Reply:
Is this resolved? Although the test name says "no leading v", the actual source starts with "v".

Reply:
🤦‍♀ 

Try now?